flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java

```diff
@@ -0,0 +1,713 @@
+    sql(
+        "INSERT INTO %s VALUES (1,'iceberg',10),(2,'b',20),(3,CAST(NULL AS VARCHAR),30)",
+        TABLE_NAME);
+    String sql = String.format("SELECT * FROM %s$history ", TABLE_NAME);
+    List<Row> result = sql(sql);
+
+    Table table = validationCatalog.loadTable(TableIdentifier.of(icebergNamespace, TABLE_NAME));
+    Iterator<Snapshot> snapshots = table.snapshots().iterator();
+    for (Row row : result) {
+      Snapshot next = snapshots.next();
+      Assert.assertEquals(
+          "Should have expected made_current_at",
+          ((Instant) row.getField(0)).toEpochMilli(),
+          next.timestampMillis());
+      Assert.assertEquals("Should have expected snapshot id", row.getField(1), next.snapshotId());
+      Assert.assertEquals("Should have expected parent id", row.getField(2), next.parentId());
+      Assert.assertEquals(
+          "Should have expected is current ancestor",
+          row.getField(3),
+          SnapshotUtil.isAncestorOf(
+              table, next.snapshotId(), table.currentSnapshot().snapshotId()));
+    }
+  }
+
+  @Test
+  public void testManifests() {
+    sql(
+        "CREATE TABLE %s (id INT, data VARCHAR,d DOUBLE) WITH ('format-version'='2', 'write.format.default'='%s')",
+        TABLE_NAME, format.name());
+    sql(
+        "INSERT INTO %s VALUES (1,'iceberg',10),(2,'b',20),(3,CAST(NULL AS VARCHAR),30)",
+        TABLE_NAME);
+    String sql = String.format("SELECT * FROM %s$manifests ", TABLE_NAME);
+    List<Row> result = sql(sql);
+
+    Table table = validationCatalog.loadTable(TableIdentifier.of(icebergNamespace, TABLE_NAME));
+    List<ManifestFile> expectedDataManifests = dataManifests(table);
+
+    for (int i = 0; i < result.size(); i++) {
+      Row row = result.get(i);
+      ManifestFile manifestFile = expectedDataManifests.get(i);
+      Assert.assertEquals(
+          "Should have expected content", row.getField(0), manifestFile.content().id());
+      Assert.assertEquals("Should have expected path", row.getField(1), manifestFile.path());
+      Assert.assertEquals("Should have expected length", row.getField(2), manifestFile.length());
+      Assert.assertEquals(
+          "Should have expected partition_spec_id",
+          row.getField(3),
+          manifestFile.partitionSpecId());
+      Assert.assertEquals(
+          "Should have expected added_snapshot_id", row.getField(4), manifestFile.snapshotId());
+      Assert.assertEquals(
+          "Should have expected added_data_files_count",
+          row.getField(5),
+          manifestFile.addedFilesCount());
+      Assert.assertEquals(
+          "Should have expected existing_data_files_count",
+          row.getField(6),
+          manifestFile.existingFilesCount());
+      Assert.assertEquals(
+          "Should have expected deleted_data_files_count",
+          row.getField(7),
+          manifestFile.deletedFilesCount());
+    }
+  }
+
+  @Test
+  public void testAllManifests() {
```

Comment:
we should create the scenario where `all_manifests` test returns different results as the `manifests` test above.

Reply:
done